### PR TITLE
DOCSP-15480 direct to mongosh docs from legacy landing page

### PR DESCRIPTION
**Pull request by davemungo**
This update is part of a series of docs changes for the new mongosh shell. 

This one redirects users from the mongo shell landing page to the mongosh page and alerts them to the mongo shell deprecation. 

Would you take a look? 

Thanks!

Current staging links: 
# Landing Page
https://docs-mongodbcom-staging.corp.mongodb.com/f2fe1da71/docs/docsworker-xlarge/DOCSP-15480-update-shell-landing-page-deprecate-mongo-shell-v5.0/mongo/

# Release Notes
https://docs-mongodbcom-staging.corp.mongodb.com/f2fe1da71/docs/docsworker-xlarge/DOCSP-15480-update-shell-landing-page-deprecate-mongo-shell-v5.0/release-notes/5.0/#new-mongosh-shell

# Compatibility Page
https://docs-mongodbcom-staging.corp.mongodb.com/f2fe1da71/docs/docsworker-xlarge/DOCSP-15480-update-shell-landing-page-deprecate-mongo-shell-v5.0/release-notes/5.0-compatibility/#shell-changes

### CODE REVIEW

File: source/mongo.txt [hunk 1]
```diff
@@ -4,417 +4,26 @@ The ``mongo`` Shell
...
-``.mongorc.js`` file *after* the JavaScript has finished processing.
-You can prevent ``.mongorc.js`` from being loaded by using the
-:option:`--norc <mongo --norc>` option.
-
-.. _mongo-shell-exit:
-
-Exit the Shell
---------------
-
-To exit the shell, type ``quit()`` or use the ``<Ctrl-C>`` shortcut.
```
Comment (addaleax) [inline source/mongo.txt:405]: Is there an equivalent for these sections for mongosh?

File: source/mongo.txt [hunk 2]
```diff
@@ -4,417 +4,26 @@ The ``mongo`` Shell
...
+   corresponding documentation for that ``MongoDB`` release.
 
-     mongo "mongodb+srv://server.example.com/"
+   *Quick Links to prior versions*
 
-  .. note::
-  
-     Use of the ``+srv`` connection string modifier automatically sets
-     the ssl option to true for the connection.
+   - :v4.4:`mongo shell v4.4 </mongo>`
```
Comment (jeff-allen-mongo) [inline source/mongo.txt:17]: Note for review: We're aware that this link is broken. It will be fixed when our platform team does their release next week.

**Review by addaleax — APPROVED** [review]

File: source/mongo.txt (hunk 1 above)
Comment (jeff-allen-mongo) [reply source/mongo.txt:405]: That would be here: [Disconnect from a Deployment](https://docs.mongodb.com/mongodb-shell/connect/#disconnect-from-a-deployment)

It's possible it should instead go here in the [mongosh Usage](https://docs.mongodb.com/mongodb-shell/run-commands/) page

Comment (jeff-allen-mongo) [reply source/mongo.txt:405]: We also have [DOCSP-15488](https://jira.mongodb.org/browse/DOCSP-15488) to audit the legacy shell material and move anything that's missing to the mongosh docs.

Comment (addaleax) [reply source/mongo.txt:405]: I was thinking about the preceding sections as well, but I think DOCSP-15488 is the right thing then :+1:

File: source/release-notes/5.0-compatibility.txt [hunk 3]
```diff
@@ -61,6 +61,15 @@ output the ``cache pressure percentage threshold`` and the
 ``current cache pressure percentage`` under
 ``wiredTiger.snapshot-window-settings``.
 
+Shell Changes
+-------------
+
+The ``mongo`` shell has been deprecated in MongoDB v5.0. The
+replacement shell is :mongosh:`mongosh </>`.
+
+The new shell is not packaged with MongoDB. To download and install
```
Comment (mmarcon) [inline source/release-notes/5.0-compatibility.txt:70]: It's not entirely true that the new shell is not packaged with MongoDB.
It's true now, but it won't be true for all platforms at the time of GA. I'll send you the link to the scope via Slack.

Essentially our expectations are:
- Windows: mongosh is **not** part of the server MSI
- Server zip/tar.gz: mongosh is **not** included
- Linux PPAs: there will be meta-packages `mongodb-{org,enterprise}` that include `mongosh`
- Homebrew: mongosh will be part of the [`mongodb-community` formula](https://github.com/mongodb/homebrew-brew/blob/master/Formula/mongodb-community.rb)

File: source/release-notes/5.0.txt [hunk 4]
```diff
@@ -186,15 +195,45 @@ for the ``"matchPrimaryNode"`` option. When set for a
...
-parameter to enable configuration of the supported cipher suites OpenSSL
-should permit when using TLS 1.3 encryption.
+New ``mongosh`` shell
+~~~~~~~~~~~~~~~~~~~~~
+
+The ``mongo`` shell has been deprecated in MongoDB v5.0. The
+replacement shell is :mongosh:`mongosh </>`. The legacy ``mongo`` shell
+will be removed in a future release.
+
+The ``monogsh`` shell is not packaged with MongoDB. See the
```
Comment (mmarcon) [inline source/release-notes/5.0.txt:210]:
```suggestion
The ``mongosh`` shell is not packaged with MongoDB. See the 
```

Comment (mmarcon) [reply source/release-notes/5.0.txt:210]: Aside from the typo, same considerations as above on the shell being packaged with MongoDB.

**Review by mmarcon — CHANGES_REQUESTED** [review]

File: source/mongo.txt (hunk 1 above)
Comment (davemungo) [reply source/mongo.txt:405]: Yes, this is just one of several expected tickets.

File: source/release-notes/5.0-compatibility.txt (hunk 3 above)
Comment (davemungo) [reply source/release-notes/5.0-compatibility.txt]: I updated the text to a more neutral pointer to the Install instructions. The changes can be made there as the packing is updated.

**Comment by davemungo** [comment]
@addaleax @mmarcon 

FYI -  I've updated the PR to reflect your suggestions. Would you take a look? 

The staging links are here: 
# Rel Notes
https://docs-mongodbcom-staging.corp.mongodb.com/64fdcf0d7/docs/docsworker-xlarge/DOCSP-15480-update-shell-landing-page-deprecate-mongo-shell-v5.0/release-notes/5.0/#shell-changes

# Compatibility
https://docs-mongodbcom-staging.corp.mongodb.com/64fdcf0d7/docs/docsworker-xlarge/DOCSP-15480-update-shell-landing-page-deprecate-mongo-shell-v5.0/release-notes/5.0-compatibility/#shell-changes

# mongo landing page
https://docs-mongodbcom-staging.corp.mongodb.com/64fdcf0d7/docs/docsworker-xlarge/DOCSP-15480-update-shell-landing-page-deprecate-mongo-shell-v5.0/mongo/#the-mongo-shell

**Comment by mmarcon** [comment]
LGTM. When we are closer to the GA I'd like to revisit this sentence on the landing page:

> The mongosh currently supports a subset of the older mongo shell methods. To maintain backwards compatibility, the methods that the new mongosh supports use the same syntax as the corresponding methods in the mongo shell.

I'd want to give it a more neutral tone so it's clear that yes, the methods/objects/commands that are supported are a subset of what was available in `mongo`, however the stuff that isn't supported was internal/undocumented/deprecated/weird and what is now in `mongosh` is all you need to work with MongoDB anyways.

**Comment by davemungo** [comment]
@mmarcon - 

Sure thing. There are a number of updates coming, no problem revisiting the wording as we get closer.

**Pull request merged**